src/TodoContex.tsx

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoType } from './enums/TodoType';
+import { RefObject } from 'react';
+
+interface TodoContextProps {
+  todos: Todo[];
+  filteredTodos: Todo[];
+  handleToggleAllButton: () => void;
+  handleSubmitButton: (event: React.FormEvent) => void;
+  title: string;
+  setTitle: (newTitle: string) => void;
+  titleField: RefObject<HTMLInputElement>;
+  completeTodo: (todoId: number) => void;
+  changedTitle: string;
+  setChangedTitle: (newTitle: string) => void;
+  deleteTodo: (todoId: number) => void;
+  handleTitleChange: (event: React.FormEvent, updatedTodo: Todo) => void;
+  completedTodosCount: number;
+  todosType: TodoType;
+  setTodosType: (todosType: TodoType) => void;
+  clearCompletedTodo: () => void;
+  changingTodo: Todo | undefined;
+  setChangingTodo: (changingTodo: Todo | undefined) => void;
+}
+
+export const TodosContext = React.createContext<TodoContextProps | undefined>(
+  undefined,
+);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [title, setTitle] = useState('');
+  const [changedTitle, setChangedTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const savedTodos = localStorage.getItem('todos');
+
+    return savedTodos ? JSON.parse(savedTodos) : [];
+  });
+  const [changingTodo, setChangingTodo] = useState<Todo | undefined>(undefined);
+  const [todosType, setTodosType] = useState<TodoType>(TodoType.all);
+  const [completedTodosCount, setCompletedTodosCount] = useState(0);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const titleField = useRef<HTMLInputElement>(null);
+  const USER_ID = 2262;
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+
+    setCompletedTodosCount(todos.filter(todo => todo.completed).length);
```

Comment:
```suggestion
    const completedTodosCount = (todos.filter(todo => todo.completed).length);
```
it just can be the var